flake8_author.py

```diff
@@ -83,8 +86,23 @@ def run(self):
             message = 'A401 __author__ attributes are not allowed'
             yield node.lineno, node.col_offset, message, type(self)
 
-        elif (node and 'pattern' in self.options and
-                not self.options['pattern'].match(node.value.s)):
-            message = ('A402 __author__ value "{0}" does not match "{1}"'
-                       .format(node.value.s, self.options['pattern'].pattern))
-            yield node.lineno, node.col_offset, message, type(self)
+        elif (node and 'pattern' in self.options):
+            if isinstance(node.value, ast.List):
+                for author in ast.literal_eval(node.value):
+                    if not self.options['pattern'].match(author):
+                        message = (
+                            'A402 __author__ value "{0}" does not match "{1}"'
+                            .format(author, self.options['pattern'].pattern)
+                        )
+                        yield (
+                            node.lineno,
+                            node.col_offset,
+                            message,
+                            type(self)
+                        )
+            elif not self.options['pattern'].match(node.value.s):
+                message = (
+                    'A402 __author__ value "{0}" does not match "{1}"'
+                    .format(node.value.s, self.options['pattern'].pattern)
+                )
+                yield node.lineno, node.col_offset, message, type(self)
```

Comment:
There's a lot of duplication here. Let's move the common bits into function.

Reply:
Moved common code to a function